Feat: Migrate upload flow

joel-medicala-yral opened the pull request:
Closes https://github.com/dolr-ai/product-roadmap/issues/191#issuecomment-2818315063 https://github.com/dolr-ai/hot-or-not-web-leptos-ssr/issues/784

uwuxyz committed:
feat: added cf upload flow

uwuxyz committed:
fix: unresolved imports

uwuxyz committed:
lint fix

uwuxyz committed:
feat: redesign upload page UI and add success screen with share option

uwuxyz committed:
Remove unused imports and clean up video upload UI layout

uwuxyz committed:
Replace map/flatten with and_then and fix string formatting in POST request

uwuxyz committed:
Improve error handling and remove share popup from video upload screen

uwuxyz committed:
Update button styles in video upload page with lighter pink color

uwuxyz committed:
Add responsive design and update post-upload flow for video uploads

rupansh-sekar-yral commented on ssr/src/page/src/upload/cf_upload.rs:
```diff
@@ -103,47 +103,47 @@ mod cf_impl {
-        user.update_post_as_ready_to_view(post_id).await?;
-        Ok(post_id)
-    }
+    // pub async fn upload_video_stream(
```
Please nuke unused code

rupansh-sekar-yral commented on ssr/src/page/src/upload/video_upload.rs:
```diff
@@ -66,6 +66,26 @@ pub fn PreVideoUpload(file_blob: WriteSignal<Option<FileWithUrl>, LocalStorage>)
+    let canister_store = auth_canisters_store();
+
+    let upload_action: Action<(), _, LocalStorage> = Action::new_local(move |_| async move {
+        let upload_base_url = "https://yral-upload-video.go-bazzinga.workers.dev";
```
This should go to consts.rs

uwuxyz committed:
Add cargo-leptos dependency and update upload URL constant
Remove unused cf_upload module and refactor video upload logic to use UPLOAD_URL constant

uwuxyz committed:
review changes
